codegen/module.go

```diff
@@ -284,18 +298,6 @@ func (system *ModuleSystem) ResolveModules(
 		if err != nil {
 			return nil, err
 		}
-
-		// Resolve dependencies again in case two instances
```

Comment:
My comment originally was that you need to _read_ the modules before you _resolve_ the modules. I thought the confusion was between the two for loops read and resolve instead of resolving in the read loop, not the fact that resolve was happening twice after read, which is clearly wrong.